python/bigquery_handler.py

```diff
@@ -0,0 +1,136 @@
+    comments = []
+    for line in code_by_lines:
+        if brackets_counter == 0 and is_method_beginning(line):
+            in_method = True
+            current_method_code_lines = []
+            count_in_method += 1
+            name = line.split("(")[0].split(" ")[-1]
+            number_of_words_in_name, name_with_spaces = get_number_of_words(name)
+            url_exists = False
+        if in_method:
+            current_method_code_lines.append(line)
+            brackets_counter += (line.count("{") - line.count("}"))
+            if not url_exists:
+                url_exists = bool(
+                    re.search(r'\.(get|navigate\(\).to)\([\w\s\+\\/]*(\"www|\"http|url|Url)[\"\w\s\+\\/]*\);', line))
+            if brackets_counter == 0:
+                # finish reading test
+                in_method = False
+                count_end_method += 1
+
+                if url_exists and number_of_words_in_name >= 5:
+                    # filtering methods
+                    instance = {'repo_name': name, 'name for description': name_with_spaces,
+                                'comments': ' '.join(comments), 'url': url_exists,
+                                'content': ' '.join(current_method_code_lines)}
+                    methods_dictionary[count_end_method] = instance
+                comments.clear()
+        else:
+            comments = update_outside_method_comments(line, comments)
+    return methods_dictionary
+
+
+def write_all_methods_in_dictionary_to_json():
+    json_file_path = project_path + '/bigquery_files/tokenized-bigquery.json'
+    with open(json_file_path, "w+", encoding='utf-8') as json_result_file:
+        for method_info in all_methods_dictionary.values():
+            json_result_file.write(json.dumps(method_info))
+            json_result_file.write('\n')
+
+
+def is_method_beginning(line):
+    """
+    Checks if a line is the beginning of a method
+    :param line: string that contains a line of code
+    :return: boolean that indicates weather the line is a beginning of a method
+    """
+    return re.match(
+        r'^[\s]*((public([\s]+))|(private([\s]+)))?(static([\s]+))?[\w<>?]+[\s]+[\w]+[\s]*\([\w ,<>?]*\)([\s]+(throws)['
+        r'\s]+[\w]*)?[\s]*[{]$',
+        line)
+
+
+def get_number_of_words(string):
+    if '_' in string:
+        # handle snake_case
+        string = string.replace('_', ' ')
+    elif string != string.lower() and string != string.upper():
+        # handle camelCase
+        string = re.sub(r'((?<=[a-z])[A-Z]|(?<!\A)[A-Z](?=[a-z]))', r' \1', string)
+    return len(string.split()), string
+
+
+def update_outside_method_comments(line, comments):
+    """
+    handles the comments variable for each out of method line
+    :param line: string that contains the line to process
+    :param comments: string that contains the current comments
+    :return: the updates comments variable
+    """
+    result = re.split(r'^[\s]*(//|@|/\*\*|/\*|\*/|\*)', line)
```

Comment:
I took a look and the comments, did you check that it extract correctly the comment if exists?
(both /** **/ and //)
BTW, it looks like must of the extracted data resembles, is most of the data in the original big query data also from the same repo? or do we have many repos?

Reply:
I checked and the comments are extracted correctly.
There are lots of different repos.